[KamiliArsyad] iP

KamiliArsyad opened the pull request:
## Meowies

> “Meow meow meowies” – My Cat ([source](thecatsite.com))

Meowies meows for your meowness. Check my website [kamiliarsyad.com](kamiliarsyad.com) to find out more!

- Cat-based
- Interactive

All you need to do is, download it from [here](https://kamiliarsyad.com).
- [x] double-click it.
- [ ] add your tasks.
- [ ] let it manage your tasks for you 😉
- [ ] And it is FREE!

Features:

1. Meows
2. Managing tasks
3. Managing deadlines
4. Reminders

In case you are curious about a box of code, here is one to satisfy your curiosity.
```java
public class Main {
    public static void main(String[] args) {
        System.out.println("meows");
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

KamiliArsyad committed:
implement name change

KamiliArsyad committed:
input handler and a chat model

KamiliArsyad committed:
implement add and list command

KamiliArsyad committed:
implement task and its status

KamiliArsyad committed:
implement different task handler (no inheritance (yet?))

KamiliArsyad committed:
added funi manual tests

KamiliArsyad committed:
Implemented wrong command and wrong format exceptions (and the handler)

KamiliArsyad committed:
Create "Wrong Command" and "Wrong Format" exception and handler

KamiliArsyad committed:
patch exception handling (to be improved later)

KamiliArsyad committed:
added tests for wrong commands/formatting

KamiliArsyad committed:
delete function

KamiliArsyad committed:
save file and added task children

KamiliArsyad committed:
modify Task by abstracting it away

KamiliArsyad committed:
implement load and save file

KamiliArsyad committed:
Merge branch 'branch-Level-7'

KamiliArsyad committed:
cleanup

KamiliArsyad committed:
stabilize date format

KamiliArsyad committed:
Merge branch 'branch-Level-8'

KamiliArsyad committed:
extracted input handler

KamiliArsyad committed:
refactor Task out

KamiliArsyad committed:
refactor TaskStorage out

KamiliArsyad committed:
add class UI

KamiliArsyad committed:
refactor UI out

KamiliArsyad committed:
refactor tasktype out

KamiliArsyad committed:
refactor and rename model, added some docs.

KamiliArsyad committed:
Merge branch 'A-MoreOOP'

KamiliArsyad committed:
Create package Duke

KamiliArsyad committed:
Merge remote-tracking branch 'origin/add-gradle-support'

KamiliArsyad committed:
add checkstyle

KamiliArsyad committed:
fix style

KamiliArsyad committed:
new task package

KamiliArsyad committed:
refactor tasks

KamiliArsyad committed:
change tasks to public

KamiliArsyad committed:
change tasks to public

KamiliArsyad committed:
add tests

KamiliArsyad committed:
configure jar

KamiliArsyad committed:
create test for taskstorage find

KamiliArsyad committed:
implement `find` command

KamiliArsyad committed:
Merge branch 'Level-9'

thaddeusong commented on src/main/java/duke/UI.java:
```diff
@@ -0,0 +1,55 @@
+            printer(input);
+        }
+
+        private void printer(String input) {
```
Probably better to use the verb "print" as the method name here.

thaddeusong commented on src/main/java/duke/task/EventTask.java:
```diff
@@ -0,0 +1,83 @@
+    }
+
+    @Override
+    protected String saveToFileString() {
```
This method sounds like it saves to the file, when it just returns the string in a save format. Maybe a name like toSaveString or toSaveFormat would be better?

dhruvir29 commented on src/main/java/duke/task/TaskStorage.java:
```diff
@@ -0,0 +1,178 @@
+    }
+
+    /**
+     * Mark a task as done.
```
The first sentence of the JavaDoc comment should start with Returns, Sends, Adds instead of Return, Send, Add. Perhaps change this to "Marks a task as done".

dhruvir29 commented on src/main/java/duke/task/TaskStorage.java:
```diff
@@ -0,0 +1,178 @@
+    }
+
+    /**
+     * Unmark a task as done.
```
Similarly, consider changing this to "Unmarks a task as done".

dhruvir29 commented on src/main/java/duke/task/TaskStorage.java:
```diff
@@ -0,0 +1,178 @@
+    }
+
+    /**
+     * Delete a task.
```
Similarly, "Deletes a task".

dhruvir29 commented on src/main/java/duke/task/TaskStorage.java:
```diff
@@ -0,0 +1,178 @@
+    }
+
+    /**
+     * Find tasks that contain the keyword.
```
Similarly, "Finds tasks that contain the keyword"

dhruvir29 commented on src/main/java/duke/InputHandler.java:
```diff
@@ -0,0 +1,61 @@
+
+import duke.task.TaskStorage;
+
+class InputHandler {
```
Perhaps add a header comment for the class InputHandler as it seems to be a requirement.

KamiliArsyad committed:
add javaFX

KamiliArsyad committed:
initialize gui

KamiliArsyad committed:
crude gui step 1

KamiliArsyad committed:
add base gui and prepare for non-cli backend

KamiliArsyad committed:
add base for non-cli backend

KamiliArsyad committed:
create all commands

KamiliArsyad committed:
fix styling

KamiliArsyad committed:
implement parser

KamiliArsyad committed:
separate cli output ui

KamiliArsyad committed:
rename UI to AppController

KamiliArsyad committed:
remove dependency to redundant InputHandler class

KamiliArsyad committed:
clean AppController

KamiliArsyad committed:
finish gui

KamiliArsyad committed:
add welcome message and handle exit in gui

KamiliArsyad committed:
delete unusued input handler class

KamiliArsyad committed:
fix jar

KamiliArsyad committed:
Fix Bug

Create a long commit message by fixing a trivially short bug.

KamiliArsyad committed:
Add Assertion

Assertion is added because assertion is life.
In fact, assertion is turing complete (I assume) and I am assertion.

KamiliArsyad committed:
Increase Code Quality

Documentation is important to document the document. This implies quality

KamiliArsyad committed:
Merge pull request #2 from KamiliArsyad/A-Assertions

Add Assertion

KamiliArsyad committed:
Merge branch 'master' into A-CodeQuality

KamiliArsyad committed:
Merge pull request #1 from KamiliArsyad/A-CodeQuality

Increase Code Quality

KamiliArsyad committed:
Add sort capability

KamiliArsyad committed:
Add sort command

KamiliArsyad committed:
Add new superclass for runtime exception

KamiliArsyad committed:
Rename exception classes

KamiliArsyad committed:
Refactor exceptions

KamiliArsyad committed:
Replace images

KamiliArsyad committed:
Add ui doc

KamiliArsyad committed:
Add meowies User Guide
